tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java

```diff
@@ -2259,6 +2261,41 @@ public void addPartitions(String databaseName, String tableName, List<CatalogPro
       CatalogUtil.closeQuietly(pstmt4);
     }
   }
+  @Override
+  public List<TablePartitionKeysProto> getAllPartitionKeys() throws CatalogException {
+    Connection conn = null;
+    Statement stmt = null;
+    ResultSet resultSet = null;
+
+    List<TablePartitionKeysProto> partitions = new ArrayList<TablePartitionKeysProto>();
+
+    try {
+      String sql = " SELECT C." + COL_PARTITIONS_PK + ", C.COLUMN_NAME, C.PARTITION_VALUE " +
+        " FROM " + TB_TABLES + " A " +
+        " JOIN " + TB_PARTTIONS + " B ON A." + COL_TABLES_PK + " = B." + COL_TABLES_PK
```

Comment:
Since it is guaranteed that the tables corresponding to partitions always exist, join looks not necessary.